File: src/walkway_control/scripts/simple_driver.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist

class SimpleDriverPy(Node):
    def __init__(self):
        # 1. Name the node
        super().__init__('simple_driver_py')

        # 2. Create the Publisher (Message Type, Topic Name, Queue Size)
        self.publisher_ = self.create_publisher(Twist, 'cmd_vel', 10)

        # 3. Create a timer to execute a callback every 0.5 seconds (2 Hz)
        self.timer = self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        # 4. Create an empty Twist message
        msg = Twist()

        # 5. Fill in the data: Drive forward at 0.5 meters per second
        msg.linear.x = 0.5
        msg.angular.z = 0.0

        # 6. Publish the message to the topic
        self.publisher_.publish(msg)
        self.get_logger().info('Publishing: Driving forward at 0.5 m/s')

def main(args=None):
    rclpy.init(args=args)       # Initialize ROS 2 communications
    node = SimpleDriverPy()     # Instantiate our node
    rclpy.spin(node)            # Keep the node alive and looping
    node.destroy_node()         # Clean up
    rclpy.shutdown()            # Shut down ROS 2

if __name__ == '__main__':
    main()

File: src/walkway_control/src/simple_driver.cpp
#include <chrono>
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

using namespace std::chrono_literals;

// Inherit from the standard rclcpp::Node
class SimpleDriverCpp : public rclcpp::Node {
    public:
    SimpleDriverCpp() : Node("simple_driver_cpp") {
        // 1. Create Publisher
        publisher_ = this->create_publisher<geometry_msgs::msg::Twist>("cmd_vel", 10);

        // 2. Create Timer
        timer_ = this->create_wall_timer(
            500ms, std::bind(&SimpleDriverCpp::timer_callback, this)
        );
    }

    private:
    void timer_callback() {
        // 3. Create and fill the message
        auto msg = geometry_msgs::msg::Twist();
        msg.linear.x = 0.5;
        msg.angular.z = 0.0;

        // 4. Publish
        publisher_->publish(msg);
        RCLCPP_INFO(this->get_logger(), "Publishing: Driving forward at 0.5 m/s");
    }

    // Declare member variables
    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char * argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SimpleDriverCpp>());
    rclcpp::shutdown();
    return 0;
}